### PR TITLE
Add mark downs and fix some typo in optimize

### PR DESCRIPTION
**Pull request by dding3**
## What changes were proposed in this pull request?
1. Add mark down files for 
   CDivTable
   Normalize
   VolumetricConvolution
   Echo
   MarginCriterion
   AddConstant
   CAdd
   Padding
   BiRecurrent
   SGD
   LeakyReLU

2. Fix typo in optimize
3. Clarify in inception readme

## How was this patch tested?
unit test

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+val module = VolumetricConvolution(nInputPlane, nOutputPlane, kT, kW, kH)
+```
+**Python:**
+```python
+module = VolumetricConvolution(n_input_plane, n_output_plane, k_t, k_w, k_h)
+```
+
+Applies a 3D convolution over an input image composed of several input planes. The input tensor
+in forward(input) is expected to be a 4D tensor (nInputPlane x time x height x width).
+ * @param nInputPlane The number of expected input planes in the image given into forward()
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: delete @

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ * Default is 0. (kT-1)/2 is often used here.
+ * @param padW The additional zeros added per width to the input planes.
+ * @param padH The additional zeros added per height to the input planes.
+ * @param withBias whether with bias
+ * @param initMethod Init method, Default, Xavier, Bilinear.
+ 
+**Scala example:**
+```scala
+val layer = VolumetricConvolution[Double](3, 4, 4, 3, 2, 3, 2, 2, 1, 1, 1)
+val input = Tensor[Double](3, 5, 6, 6).rand()
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: yiheng says it is better to fix input  content, not rand

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+-0.03242712780702128    -0.3649814573965776     -0.041354367554219396
+-0.093357235374353      -0.48948518373955563    -0.4277957331255818
+-0.13937510329639902    -0.3578734836823814     -0.449763889250075
+-0.22252364458108947    ...
+```
+
+**Python example:**
+```python
+layer = VolumetricConvolution(3, 4, 4, 3, 2, 3, 2, 2, 1, 1, 1)
+input = np.random.rand(3,5,6,6)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: no forward????

File: docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```scala
+criterion = MarginCriterion()
+```
+**Python:**
+```python
+criterion = MarginCriterion()
+```
+
+Creates a criterion that optimizes a two-class classification hinge loss (margin-based loss) between input x (a Tensor of dimension 1) and output y.
+ * @param margin if unspecified, is by default 1.
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md]: no @

File: docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+## MarginCriterion ##
+
+**Scala:**
+```scala
+criterion = MarginCriterion()
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md]: criterion = MarginCriterion(margin) ?

File: docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 3x2]
+
+criterion.forward(input, target)
+res0: Double = 0.8222855331373905
+```
+
+**Python example:**
+```python
+criterion = MarginCriterion()
+input = np.random.rand(3, 2)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MarginCriterion.md]: perhaps don't rand

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/LeakyReLU.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+```python
+layer = LeakyReLU()
+```
+
+It is a transfer module that applies LeakyReLU, which parameter
+negval sets the slope of the negative part:
+ LeakyReLU is defined as:
+  f(x) = max(0, x) + negval * min(0, x)
+
+ * @param negval sets the slope of the negative partl, default is 0.01
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/LeakyReLU.md]: no @

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
+## VolumetricConvolution ##
+
+**Scala:**
+```scala
+val module = VolumetricConvolution(nInputPlane, nOutputPlane, kT, kW, kH)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: Is it need to add [T]?

File: docs/docs/APIdocs/layers/nn/Normalization Layers/Normalize.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+## Normalize ##
+
+**Scala:**
+```scala
+val module = Normalize(p)
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Normalization Layers/Normalize.md]: whole parameters

File: docs/docs/APIdocs/layers/nn/Recurrent Layers/BiRecurrent.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+This layer implement a bidirectional recurrent neural network
+ * @param merge concat or add the output tensor of the two RNNs. Default is add
+
+**Scala example:**
+```scala
+val module = BiRecurrent[Double](CAddTable[Double]())
+.add(RnnCell[Double](6, 4, Sigmoid[Double]()))
+val input = Tensor[Double](Array(1, 2, 6)).rand()
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Recurrent Layers/BiRecurrent.md]: rand input, not clear

File: docs/docs/APIdocs/layers/optim/SGD.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+val x = Tensor[Double](1)
+x.fill(10)
+val config = T("learningRate" -> 1e-3)
+for (i <- 1 to 10) {
+  optimMethod.optimize(feval, x, config, config)
+}
+println("x after optimize: " + x)
+x after optimize: 9.801790433519495
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcD$sp of size 1]
+```
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/optim/SGD.md]: where is the python example

**Comment by zhangxiaoli73** [comment]
Besides, I'm not sure whether to add backward

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md (hunk 3 above)
Comment (dding3) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: Added

File: docs/docs/APIdocs/layers/optim/SGD.md (hunk 11 above)
Comment (dding3) [inline docs/docs/APIdocs/layers/optim/SGD.md]: Added

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+ * @param dH The step of the convolution in the height dimension. Default is 1
+ * @param padT Additional zeros added to the input plane data on both sides of time axis.
+ * Default is 0. (kT-1)/2 is often used here.
+ * @param padW The additional zeros added per width to the input planes.
+ * @param padH The additional zeros added per height to the input planes.
+ * @param withBias whether with bias.
+ 
+**Scala example:**
+```scala
+val layer = VolumetricConvolution[Float](2, 4, 4, 3, 2, dT=1, dW=1, dH=1,
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: please import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric.NumericFloat, then you don't have to appoint "Float"

File: docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+-0.031611152    -0.021568365    -0.24138525     -0.46093956
+
+(3,1,.,.) =
+0.25616848      0.6713682       0.28883895      0.31497225
+0.3645328       0.3806299       0.5575222       0.22212546
+0.49616873      0.18132609      0.5415702       0.5869311
+0.65951777      0.41917348      0.27746195      0.46284527
+0.5698555       0.2887269       0.6030495       0.35626367
+
+(4...
```
Comment (zhangxiaoli73) [inline docs/docs/APIdocs/layers/nn/Convolution Layers/VolumetricConvolution.md]: not complete

**Comment by zhangxiaoli73** [comment]
LGTM

**Pull request merged**